page_analyzer/app.py

```diff
@@ -1,8 +1,76 @@
-from flask import Flask, render_template
+import os
+from typing import Any, Dict
+
+from dotenv import load_dotenv
+from flask import Flask, flash, redirect, render_template, request, url_for
+from page_analyzer.utils.db import connect_db
+from page_analyzer.utils.url import normalize, validate
+
+load_dotenv()
+
 
 app = Flask(__name__)
+app.secret_key = os.getenv("SECRET_KEY")
+
+
+def get_url_context(url_id: int) -> Dict[str, str]:
+    conn = connect_db()
+    cur = conn.cursor()
+    cur.execute("SELECT name, created_at FROM urls WHERE id='%s';", (url_id,))
+    page_url, created_at = cur.fetchone()
+    cur.close()
+    conn.close()
+    context = {"id": str(url_id), "name": str(page_url), "created_at": str(created_at)}
+    return context
 
 
 @app.route("/")
-def hello_world() -> str:
+def index() -> str:
     return render_template("index.html")
+
+
+@app.route("/url", methods=["POST"])
+def post_url() -> Any:
```

Comment:
it's a good practice to give endpoint functions reasonable names like `create_url` or smth